uint16[] angles_w_objects
